src/compiler/checker.ts

```diff
@@ -30499,7 +30525,6 @@ namespace ts {
             }
             const file = getSourceFileOfNode(node);
             Debug.assert(!!(file.flags & NodeFlags.PossiblyContainsImportMeta), "Containing file is missing import meta node flag.");
-            Debug.assert(!!file.externalModuleIndicator, "Containing file should be a module.");

```

Comment:
This is only true when the import meta property is a full `import.meta`, but not a halfway `import.me` or something like that, which is still considered a meta property.